### PR TITLE
feat: Add JSON schema for `configtype.Duration`

### PR DESCRIPTION
**Pull request by candiduslynx**

### CODE REVIEW

File: configtype/duration.go [hunk 1]
```diff
@@ -19,6 +21,15 @@ func NewDuration(d time.Duration) Duration {
 	}
 }
 
+func (Duration) JSONSchema() *jsonschema.Schema {
+	// provide a pattern for only non-negative durations
```
Comment (hermanschaaf) [inline configtype/duration.go:25]: The regex doesn't look like it only supports non-negative durations?

Comment (candiduslynx) [reply configtype/duration.go:25]: yeah, I'll remove the comment (as we do support negative durations even in tests)

Comment (candiduslynx) [reply configtype/duration.go]: removed in df0b27e

File: configtype/duration_test.go [hunk 2]
```diff
@@ -52,3 +56,74 @@ func TestComparability(t *testing.T) {
...
+			Err:  true,
+			Spec: `false`,
+		},
+		{
+			Name: "bad format",
+			Err:  true,
+			Spec: `false`,
+		},
+	},
+		func() []testCase {
```
Comment (hermanschaaf) [inline configtype/duration_test.go:96]: nit: this syntax of appending a function that calls itself and appends the results in-line feels unnecessarily complicated, it'd be more readable to introduce a variable first and append that, like:

```
generatedCases := make([]testcase, 20)
for i := 0; i < 20; i++ {
     generatedCases[i] = generateTestCase()
}

...
cases := append([]testCase{
		{
			Name: "empty",
			Err:  true,
			Spec: `""`,
		},
                ...,
                generatedCases...
}
```

or have a few manually created test cases, there doesn't seem to be enough complexity here to warrant generating random cases

**Review by hermanschaaf — APPROVED** [review]

**Pull request merged**